icechunk-python/tests/test_virtual_ref.py

```diff
@@ -104,6 +137,17 @@ async def test_write_minio_virtual_refs():
     assert "s3://testbucket/path/to/python/chunk-1" in all_locations
     assert "s3://testbucket/path/to/python/chunk-2" in all_locations
 
+    # check we cannot get refs with bad etag o that were modified
+    import icechunk
+    with pytest.raises(IcechunkError, match="chunk has changed"):
+        await store.get("c/2/0/0", prototype=buffer_prototype)
+    with pytest.raises(IcechunkError, match="chunk has changed"):
+        await store.get("c/3/0/0", prototype=buffer_prototype)
+    with pytest.raises(IcechunkError, match="chunk has changed"):
```

Comment:
We should make a specific error type for this that inherits from `IcechunkError`

Reply:
yeah